### DISCUSSION TITLE
Can LLaMa learn from interactions?

### DISCUSSION THREAD

**Discussion by nazthelizard122** [ggml-org/llama.cpp#541] [Q&A] [answered]
Can LLaMa learn from interactions? ive talked with it a lot and it doesn't seem like it remembers our previous interactions
I could modify the chat_with_bob.txt and add all of our interactions but that would increase the load time substantially

**Comment by anzz1** [comment, marked as answer]
Nope, the model is immutable and sessions do not alter it whatsoever, so it cannot remember any details from previous interactions.

**Comment by x02Sylvie** [comment]
Not at the moment, but it may be possible in future to limited extent with LTM (Long-Term Memory) implementations such as this extension for text generation webui (https://github.com/wawawario2/text-generation-webui)

**Reply by nazthelizard122** [reply]
Seems like that just modifies the first prompt

**Reply by xor2003** [reply]
It does not look like this for me

**Comment by FNsi** [comment]
I think it can learn, but after you close it, no.
